client/components/NavigationBar.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Modal } from 'react-materialize';
+import { logout } from '../actions/loginActions';
+import SideNavBar from './SideNavBar';
+import { searchDocuments, searchUserDocuments }
+from '../actions/documentsActions';
+import { searchUsers } from '../actions/usersActions';
+import { searchRoles } from '../actions/rolesActions';
+
+class NavigationBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = this.onChange.bind(this);
+    this.logout = this.logout.bind(this);
+  }
+
+  logout(e) {
+    e.preventDefault();
+    this.props.logout();
+  }
+
+  onChange(e) {
```

Comment:
I think it will be better if the component representing each of these routes handle their search. This long chain of `if-else` indicates a code smell